src/test/suite/didactUriCompletionItemProvider.test.ts

```diff
@@ -0,0 +1,197 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import { expect } from 'chai';
+import { SnippetString } from 'vscode';
+import { DidactUriCompletionItemProvider, DIDACT_COMMAND_PREFIX } from "../../didactUriCompletionItemProvider";
+import { getContext } from '../../extensionFunctions';
+import * as vscode from 'vscode';
+import * as path from 'path'
+
+suite("Didact URI completion provider tests", function () {
+
+	const ctx = getContext();
+	const provider = new DidactUriCompletionItemProvider(ctx);
+	
+	test("that all commands in the didactCompletionCatalog.json are available", async () => {
+		const catalog : any = provider.getCompletionCatalog(ctx);
+		const vsCommands : string[] = await vscode.commands.getCommands(true);
+
+		suite('walk through each provided completion', () => {
+			catalog.forEach(function(completion: { fullCommandId: string; }){
+				const fullCommandId = completion.fullCommandId;
+				test(`the command ${fullCommandId} should exist in the vscode commands list`, () => {
+					expect(vsCommands.includes(fullCommandId)).to.be.true;
+				});
+			});
+		});
+	});
+
+	test("that the didact protocol completion returns with didact://?commandId=", () => {
+		const completionItem = provider.didactProtocolCompletion();
+		expect(completionItem).to.not.be.null;
+		expect(completionItem).to.not.be.undefined;
+		expect(completionItem.insertText).to.be.equal(DIDACT_COMMAND_PREFIX);
+	});
+
+	test("that the match utility returns expected results for simple didact uri", () => {
+		const match = provider.findMatchForCommandVariable('didact://?commandId=vscode.didact.');
+		expect(match).to.not.be.null;
+		if (match) {
+			expect(match[0]).to.be.equal('?commandId=vscode.didact.');
+			expect(match[1]).to.be.equal('vscode.didact.');
+		}
+	});
+
+	test("that the match utility returns expected results for didact uri with full command and properties", () => {
+		const match = provider.findMatchForCommandVariable('didact://?commandId=vscode.didact.closeNamedTerminal&text=NamedTerminal');
+		expect(match).to.not.be.null;
+		if (match) {
+			expect(match[0]).to.be.equal('?commandId=vscode.didact.closeNamedTerminal');
+			expect(match[1]).to.be.equal('vscode.didact.closeNamedTerminal');
+		}
+	});
+
+	test("that the command processing for a command prefix returns expected results", async () => {
+		const match = provider.findMatchForCommandVariable('didact://?commandId=vscode.didact.');
+		const completionList = await provider.processCommands(match);
+		expect(completionList.items.length).to.be.equal(26);
```

Comment:
can be simplified with `expect(completions.items).to.have.lengthOf(26)` it will provide a more precise error message too.